newsroom/topics/topics.py

```diff
@@ -81,17 +81,12 @@ def on_update(self, updates, original):
 
         # If ``is_global`` has been turned off, then remove all subscribers
         # except for the owner of the Topic
-        if (
-            original.get("is_global")
-            and "is_global" in updates
-            and not updates.get("is_global")
-            and not updates.get("subscribers")
-        ):
+        if original.get("is_global") and "is_global" in updates and not updates.get("is_global"):
             # First find the subscriber entry for the original user
             subscriber = next(
                 (
                     subscriber
-                    for subscriber in original.get("subscribers", [])
+                    for subscriber in updates.get("subscribers", [])
```

Comment:
not sure it will work now if subscribers are not updated, imo should be like:
```py
for subscriber in (updates.get("subscribers") or original.get('subscribers") or []):
```

Reply:
I added this :- `for subscriber in updates.get("subscribers", original.get("subscribers", []))`

Reply:
I see, it's not the same because if you would have `updates["subscribers"] = None` it would raise an error

Reply:
correct